### PR TITLE
actions: only run builds on pull request

### PR DESCRIPTION
**Pull request by gdams**
currently, we run builds on code that has been merged into master. This is a waste of executor time

### CODE REVIEW

**Comment by aahlenst** [comment]
I don't think that's a good idea. We don't require PRs to be updated before we merge them. Can you enable all branches instead? This would allow to run the tests on forks and that do not count towards our limits. So people could develop in their accounts and do not have to push WIP PRs just to get the PR tester to run.

**Comment by gdams** [comment]
@aahlenst by the time the PR has been merged it's too late anyway. It's not like anybody actually monitors the branch builds. I'd rather require branches be updated with master before being merged

**Review by M-Davies — CHANGES_REQUESTED** [review]
Regardless whether this is a good idea or not, we probably need to be documenting the pr tester better (all of our code actually 😅). Can you hold off on this until https://github.com/AdoptOpenJDK/openjdk-build/pull/2216 is merged and then add a note in there of these changes?

**Comment by aahlenst** [comment]
> It's not like anybody actually monitors the branch builds.

That can be automated away, for example by notifying the build channel if master breaks.

> I'd rather require branches be updated with master before being merged.

I haven't found an option to enforce this. Furthermore, this would induce and even bigger backlog because mergeable PRs have to be rebased and retested after ever merge to the master branch.

**Review by karianna — APPROVED** [review]
LGTM - I agree with George's premise here.

**Review by M-Davies — APPROVED** [review]
LGTM now. The documentation I was referencing was designed and merged with this PR in mind

**Pull request merged**
